client/lib/purchases/index.ts

```diff
@@ -119,13 +126,20 @@ function getSitePurchasesStatus( site: SiteWithPurchases ) {
 	if ( purchases.some( ( purchase ) => purchase.currentPurchaseStatus === 'expiringSoon' ) ) {
 		return 'hasExpiringSoon';
 	}
+	if ( ! site.isConnected && site.slug?.startsWith( 'siteless.akismet.com' ) ) {
+		return 'sitelessPlan'; // i.e. - Akismet plans
+	}
 
 	return 'noPaymentActionNeeded';
 }
 
-function getPurchaseStatus( purchase: PurchaseWithStatus ) {
+function getPurchaseStatus( purchase: PurchaseWithStatus, userId?: number ) {
 	const expiry = moment( purchase.expiryDate );
 
+	if ( userId && purchase.userId !== userId && purchase?.purchaserId === userId ) {
```

Comment:
there is no need for optional chaining in purchaserId - if it's null, it's gonna crash with userId :D also, I see below that we treat `purchase` as defined here, and indeed existing Typescript typing suggests the same.